rasa/utils/common.py

```diff
@@ -407,7 +407,7 @@ async def call_potential_coroutine(
     Returns:
         The return value of the function.
     """
-    if asyncio.iscoroutine(coroutine_or_return_value):
+    if inspect.iscoroutine(coroutine_or_return_value):
```

Comment:
this makes a difference, but I think that we want `inspect`'s version here

Reply:
This is also consistent with what will be introduced in https://github.com/RasaHQ/rasa/pull/10696